src/service/filtering/analyzers.py

```diff
@@ -36,9 +36,16 @@
     )
 }
 
-def get_analyzer(strategy: FilterStrategy | None) -> str:
-    """ Get the name of the appropriate ArangoDB analyzer to use for a given filter strategy. """
-    return _COL2ANALYZER.get(strategy, DEFAULT_ANALYZER)
+def get_analyzer(strategy: FilterStrategy | None, return_none_for_default_analyzer: bool = False
+    ) -> str:
+    """
+    Get the name of the appropriate ArangoDB analyzer to use for a given filter strategy.
+    
+    strategy - the strategy to translate to an analyzer
+    return_none_for_default_analyzer - rather than returning the default analyzer, return None.
+    """
+    df = None if return_none_for_default_analyzer else DEFAULT_ANALYZER
```

Comment:
can we rename `df` to `default_analyzer`?  It's a common convention to use df for dataframes. Or we can just do `return _COL2ANALYZER.get(strategy, None if return_none_for_default_analyzer else DEFAULT_ANALYZER)`

Reply:
yeah... if pandas were used in this module I'd agree, but not using any variables that are frequently used in entirely different contexts doesn't seem reasonable to me. Means we shouldn't use np either, for example.